server/models/chan.ts

```diff
@@ -63,20 +62,22 @@ class Chan {
 	static optionalProperties = ["userAway", "special", "data", "closed", "num_users"];
 
 	constructor(attr?: Partial<Chan>) {
-		_.defaults(this, attr, {
-			id: 0,
-			messages: [],
-			name: "",
-			key: "",
-			topic: "",
-			type: ChanType.CHANNEL,
-			state: ChanState.PARTED,
-			firstUnread: 0,
-			unread: 0,
-			highlight: 0,
-			users: new Map(),
-			muted: false,
-		});
+		this.id = 0;
+		this.messages = [];
+		this.name = "";
+		this.key = "";
+		this.topic = "";
+		this.type = ChanType.CHANNEL;
+		this.state = ChanState.PARTED;
+		this.firstUnread = 0;
+		this.unread = 0;
+		this.highlight = 0;
+		this.users = new Map();
+		this.muted = false;
+
+		if (attr) {
+			Object.assign(this, attr);
```

Comment:
I think a better fix would be make all of the parameters that chan creation uses explicit parameters of the constructor, such as `constructor(name, type, key, muted)` (and whatever else parameters can be set on creation.

Alternatively, it could be static helpers such as `createNewQuery` which returns an instance of new `Chan` that is set to type of query.